### PR TITLE
Adding Visual Studio 2013 support

### PR DESCRIPTION
**Pull request by JeremyLWright**
* CMakeLists.txt: Adding cmake to allow simple compilation with visual studio.
* include/autocheck/generator.hpp: Removed the explicit initializer since visual studio cannot compile variadic templates like this, however it seems this is the default behavior anyway.
* include/autocheck/value.hpp: Removed the union at the cost of some memory since Visual Studio cannot compile unrestricted unions yet.
* test/check.cpp: Moved the failing test to a separate test to make the failure more explicit and describe the indent.
* test/generator_combinators.cpp: Added explicit capture of the factor.

### CODE REVIEW

File: include/autocheck/generator.hpp [hunk 1]
```diff
@@ -271,8 +271,12 @@ namespace autocheck {
       std::tuple<Gens...> gens;
 
     public:
-      cons_generator() :
-        gens(Gens()...) {}
+      cons_generator() 
+#ifndef _MSC_VER
+		  :
```
Comment (thejohnfreeman) [inline include/autocheck/generator.hpp]: No tabs please. Spaces only. Ok to put colon and initializer list on same line.

File: test/check.cpp [hunk 2]
```diff
@@ -26,16 +26,20 @@ void insert_sorted(const T& x, std::vector<T>& xs) {
   }
 }
 
-TEST(Check, Compiles) {
+TEST(Check, DemonstrateFailingTest)
+{
```
Comment (thejohnfreeman) [inline test/check.cpp]: Put brace on same line as TEST macro.

File: include/autocheck/generator.hpp (hunk 1 above)
Comment (thejohnfreeman) [inline include/autocheck/generator.hpp]: To be clear, you might need to expand tabs in the other edits; I'm not going to tag each one.

**Comment by thejohnfreeman** [comment]
What happened here? I'm sorry if it appeared I wasn't responsive; I didn't get any email about the most recent fixes. I'd like to review and accept these changes if you're still willing to see them through.

**Comment by JeremyLWright** [comment]
Hello,
Thank you for reaching out. I do want to see the additions merged back in. I think you built a fantastic library. I am enjoying using it. I wrote an article on my blog using your library (http://www.codestrokes.com/2015/07/property_driven_design_minmax/) I plan to write a series of them, and fold it into the autocheck documentation/tutorial.  
I also added support for the catch C++ testing library (https://github.com/philsquared/Catch). I hacked it in though. Let me clean up its integration, and be sure it compiles on clang and gcc. Then I'll make a new merge request. 
Sincerely,Jeremy
Date: Wed, 8 Jul 2015 05:23:47 -0700

**Comment by thejohnfreeman** [comment]
Awesome, I'm looking forward to it!

**Pull request closed without merging**